### ISSUE TITLE
cloc reports Kotlin code as comments when executed in parallel

### ISSUE THREAD

**Issue by lberdiales** [AlDanial/cloc#738]
When running `cloc` in parallel for our codebase, a couple files are reported with 0 lines of code, inspecting the `cloc` output for those specific files we discovered that the file's code has been interpreted as comments.

```
cloc CameraFragment.kt --include-ext=kt --json

{"header" : {
  "cloc_url"           : "github.com/AlDanial/cloc",
  "cloc_version"       : "1.96",
  "elapsed_seconds"    : 1.12588787078857,
  "n_files"            : 1,
  "n_lines"            : 441,
  "files_per_second"   : 0.88818791457412,
  "lines_per_second"   : 391.690870327187},
"Kotlin" :{
  "nFiles": 1,
  "blank": 41,
  "comment": 400,
  "code": 0},
"SUM": {
  "blank": 41,
  "comment": 400,
  "code": 0,
  "nFiles": 1} }
```

**cloc; OS; OS version**
 - cloc version: 1.96
 - OS: macOS
 - OS version: Monterey 12.6

**To Reproduce**
Couldn't create an scenario to reproduce the issue yet.

**Expected result**
File's code & comments are interpreted correctly.

**Additional context**
We are sure this is related to parallel execution as this only happens when we run the codebase analysis in parallel, if we run same analysis but file by file (serial execution) the lines of code are reported correctly.
Here is the output of an isolated execution of `cloc` for the same file.
```
{"header" : {
  "cloc_url"           : "github.com/AlDanial/cloc",
  "cloc_version"       : "1.96",
  "elapsed_seconds"    : 0.319792985916138,
  "n_files"            : 1,
  "n_lines"            : 441,
  "files_per_second"   : 3.12702293058497,
  "lines_per_second"   : 1379.01711238797},
"Kotlin" :{
  "nFiles": 1,
  "blank": 41,
  "comment": 0,
  "code": 400},
"SUM": {
  "blank": 41,
  "comment": 0,
  "code": 400,
  "nFiles": 1} }
```

**Comment by AlDanial** [comment]
@stsnel : can you take a look this one after @lberdiales describes a way to reproduce?

**Comment by stsnel** [comment]
@AlDanial yes, sure :+1: I expect to be able to look into it within roughly a week.

**Comment by stsnel** [comment]
I have tried to reproduce this issue on a macOS Ventura (13.4.1c) system by running cloc 1.96 twenty times with  parameters `--include-ext=kt --json` and 20 times with parameters `--include-ext=kt --json --processes=4` on a local clone of the kotlinx.coroutines repository (https://github.com/Kotlin/kotlinx.coroutines). I got the same number of comment lines and code lines each time, irrespective of whether multiprocessing was enabled using the `--processes=n` option.

@lberdiales When you say the issue is related to parallel execution, are you referring to using the `--processes=n` option with n more than 1? Is it possible to share a link to the source code where this issue was detected?

**Comment by AlDanial** [comment]
Closing this as not reproducible.
